src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndex.cs

```diff
@@ -242,6 +242,65 @@ public override int HandleMap(IndexItem indexItem, IEnumerable mapResults, Lazy<
             }
         }
 
+        private void UpdateMapFields()
+        {
+            if (_isFanout == false)
+                return;
+
+            // we have a fanout in the map reduce index, there are 2 options:
+            // the reduce result has the same OR a different path as one of the reduce keys.
+            // if it's the same path, we need to insert each value separately in the mapped results.
+            // if it isn't, we need to calculate the result for each entry in the fanout (will have the same result).
+            //
+            // Examples:
+            //
+            // The SUM has the same path as one of the reduce keys
+            // from 'Messages'
+            // group by Name, Data[].Items[].ProductName
+            // select Name, Data[].Items[].ProductName as ProductName, sum(Data[].Items[].TotalPrice) as Total
+            // 
+            // The SUM has a different path than any of the reduce keys
+            // from 'Messages'
+            // group by Name, Tries[].ResultMessage
+            // select Name, Tries[].ResultMessage as ResultMessage, sum(Data[].Items[].TotalPrice) as Total
+
+            var groupByFieldsPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            
+            foreach ((_, AutoIndexField value) in Definition.GroupByFields)
+            {
+                if (value.GroupByArrayBehavior != GroupByArrayBehavior.ByIndividualValues)
+                    continue;
+
+                var lastIndexOfGroupByField = value.Name.LastIndexOf("[]", StringComparison.OrdinalIgnoreCase);
```

Comment:
Please double check that but I think the usage of `[]` is optional. So `Tries[].ResultMessage` is the same as `Tries.ResultMessage`.

Reply:
There is a difference. It's handled in the `BlittableJsonTraverser`.
If we have a `PropertySeparator` (.) we try to read it as `BlittableJsonReaderObject`.
If we have a `CollectionSeparatorStartBracket` ([) we try it as `BlittableJsonReaderArray`.

Reply:
So we're interested only in `[]` because we're doing it just for fanout indexes. Makes sense.

Reply:
Although originally you had only the check for `.`, didn't you?

Reply:
Yes. I'm looking for the prefix before `[].`
Using just `.` doesn't support the last case in the test:
```
from 'Messages' " +
                          "group by Name, Numbers[] " +
                          "select Name, Numbers[] as OriginalNumber, sum(Numbers) as Total
```

Reply:
OK. Thanks